### PR TITLE
Finalized StopEstimate-internal model and improved mock-data generation

### PR DESCRIPTION
**Pull request by paasovaara**
- finalized StopEstimate data model
- added lots of mock data generation functions
- fixed a bug in test framework, read events until they stop so we can catch errors if some message is delivered even though it was not expected.

### CODE REVIEW

File: src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java [hunk 1]
```diff
@@ -112,4 +124,152 @@ public static void setIsViaPoint(PubtransTableProtos.Common.Builder builder, boo
...
+
+    public static PubtransTableProtos.ROIArrival mockROIArrival(long dvjId, String routeName, int joreDirection, int stopSequence, long targetDateTimeMs) throws Exception {
+        PubtransTableProtos.Common common = generateValidCommon(dvjId, stopSequence, targetDateTimeMs).build();
+        PubtransTableProtos.DOITripInfo info = mockDOITripInfo(dvjId, routeName, joreDirection);
+        return mockROIArrival(common, info);
+    }
+
+    public static PubtransTableProtos.ROIArrival mockROIArrival(PubtransTableProtos.Common common, PubtransTableProtos.DOITripInfo info) throws Exception {
+        PubtransTableProtos.ROIArrival.Builder builder = PubtransTableProtos.ROIArrival.newBuilder();
+        builder.setSchemaVersion(builder.getSchemaVersion());
```
Comment (kenkus-futurice) [inline src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java]: What does this do?

File: src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java [hunk 2]
```diff
@@ -112,4 +124,152 @@ public static void setIsViaPoint(PubtransTableProtos.Common.Builder builder, boo
...
+    }
+
+    public static PubtransTableProtos.ROIDeparture mockROIDeparture(PubtransTableProtos.Common common, PubtransTableProtos.DOITripInfo info) throws Exception {
+        PubtransTableProtos.ROIDeparture.Builder builder = PubtransTableProtos.ROIDeparture.newBuilder();
+        builder.setSchemaVersion(builder.getSchemaVersion());
+        builder.setCommon(common);
+        builder.setTripInfo(info);
+
+        builder.setHasDestinationDisplayId(0);
+        builder.setHasDestinationDisplayId(0);
```
Comment (kenkus-futurice) [inline src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java]: What is this set twice?

File: src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java (hunk 1 above)
Comment (paasovaara) [reply src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java]: we add the default-value for the schema in the protobuf file. syntax:

```   
required int32 SchemaVersion = 1 [default = 1];
```

this already sets the value to be 1 for the getSchemaVersion(), but the generated java code however is stupid and doesn't actually raise the required-flag for these default fields, so when we call .build() -> throws exception, required field not set.  therefore we still need to explicitly set it to raise the required-flag

File: src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java (hunk 2 above)
Comment (paasovaara) [reply src/integration-test/java/fi/hsl/common/transitdata/MockDataUtils.java]: good catch, copy-paste error!

**Pull request merged**